pkg/apis/serving/v1alpha1/llm_inference_service_types_func.go

```diff
@@ -18,8 +18,16 @@ package v1alpha1
 
 import (
 	"k8s.io/utils/ptr"
+	"knative.dev/pkg/kmeta"
 )
 
+func (r *RouterSpec) EPPServiceName(llmSvc *LLMInferenceService) string {
+	if r == nil || r.Route == nil || r.Scheduler == nil || r.Scheduler.Pool == nil || !r.Scheduler.Pool.HasRef() || r.Scheduler.Pool.Spec == nil || r.Scheduler.Pool.Spec.ExtensionRef == nil {
```

Comment:
Can you add comment here why there are so many checks

Reply:
+1 
Have a higher level function to check this. Not all checks can be nil independently

Reply:
Essentially what this is trying to do is to figure out whether the EPPServiceName is overriden in the `LLMInferenceService`, all this checks ensure we use the "default EPP Service" if it's not overriden 

To override it, users can specify `r.Scheduler.Pool.Spec.ExtensionRef.Name` and all nested fields can be `nil`